whiteout/resolver.go

```diff
@@ -0,0 +1,94 @@
+package whiteout
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+
+	"github.com/quay/zlog"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/indexer"
+)
+
+var (
+	_ indexer.Resolver = (*Resolver)(nil)
+)
+
+type Resolver struct{}
+
+func (r *Resolver) Resolve(ctx context.Context, ir *claircore.IndexReport, layers []*claircore.Layer) *claircore.IndexReport {
+	// Here we need to check if any of the packages
+	// found are moot due to whiteouts.
+	ls := newLayerSorter(layers)
+	finalPackages := map[string]*claircore.Package{}
+	finalEnvironments := map[string][]*claircore.Environment{}
+	for pkgID, pkg := range ir.Packages {
+		packageDeleted := false
+		// First env.IntroducedIn?
+		packageLayer := ir.Environments[pkgID][0].IntroducedIn.String()
```

Comment:
this is guaranteed to be sorted by layer?

Reply:
This is a good point, I'm updating this to be newest layer that has the package as it could feasibly have been installed, deleted, and re-installed.